dataflow/pkg/scheduler/lifecycle/lifecycle_sched.go

```diff
@@ -194,34 +197,39 @@ func checkTransitionValidation(source int32, destination int32) bool {
 	return true
 }
 
-func getObjects(r *InternalLifecycleRule, offset, limit int32) ([]*osdss3.Object, error) {
+func getObjects(r *InternalLifecycleRule, marker string, limit int32) ([]*osdss3.Object, error) {
 	// Get objects by communicating with s3 service.
 	filt := make(map[string]string)
-	if len(r.Filter.Prefix) > 0 {
-		filt[KObjKey] = "^" + r.Filter.Prefix
-	}
 
-	modifyFilt := fmt.Sprintf("{\"gte\":\"%d\"}", r.Days)
-	filt[KLastModified] = modifyFilt
+	timeFilt := fmt.Sprintf("{\"lte\":\"%s\"}",
+		time.Now().AddDate(0,0,int(0-r.Days)).Format(TIME_LAYOUT_TIDB))
+	filt[KLastModified] = timeFilt
 	if r.ActionType != ActionExpiration {
 		filt[KStorageTier] = strconv.Itoa(int(r.Tier))
 	}
 
+	log.Infof("filt: %+v\n", filt)
 	s3req := osdss3.ListObjectsRequest{
```

Comment:
Using uppercase on first letter, and format this file

Reply:
Done